cpp/BUILD.bazel

```diff
@@ -82,19 +81,50 @@ cc_test(
         "ray_api",
         "@com_google_googletest//:gtest_main",
     ],
+    data = [
+        "cluster_mode_test.so",
+    ],
+    args = [
+        "$(location cluster_mode_test.so)",
+    ]
+)
+
+cc_binary(
+    name = "cluster_mode_test.so",
```

Comment:
I'm not 100% sure, but what's the motivation behind this being a `cc_binary` named `x.so` vs a `cc_library` (implicitly a `.so`). I think the static linking and everything works the same, except it'd be more consistent.

If it's to do with compiling in `gtest_main, then that would suggest a third target -- one for the shared library (that ultimately will be distributed around the cluster), one for the test, and one for the default worker

Reply:
I don't need a static library of `cluster_mode_test `, but cc_library has no option to control this?

Reply:
https://docs.bazel.build/versions/master/be/c-cpp.html#cc_binary.linkstatic